src/ui/public/state_management/state_monitor.js

```diff
@@ -50,9 +55,22 @@ function stateMonitor(state, defaultState) {
   };
 
   return {
+    setDefaultState(customInitialState) {
+      // check the current status
+      const currentStatus = getStatus();
```

Comment:
Can we rename this to `previousStatus` since it's used to be compared with the real current states on line 68? And we can be move it below the parameter-checking? I think it makes sense to do parameter-checking first, before any other logic, since if an error is thrown the logic is pointless.